src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -3007,11 +3007,15 @@ public static OperationStatus FromHexString(ReadOnlySpan<char> source, Span<byte
                 quotient = destination.Length;
                 result = OperationStatus.DestinationTooSmall;
             }
-            else if (remainder == 1)
+            else if (destination.Length > quotient)
             {
-                source = source.Slice(0, source.Length - 1);
-                destination = destination.Slice(0, destination.Length - 1);
+                if (remainder == 1)
+                {
+                    source = source.Slice(0, source.Length - 1);
+                }
+
                 result = OperationStatus.NeedMoreData;
```

Comment:
If the destination buffer can fit the result and the remainder is 0, why should the status be `NeedMoreData`?